src/coprocessor/endpoint.rs

```diff
@@ -423,17 +423,16 @@ impl<E: Engine> Endpoint<E> {
         let mut tracker = Box::new(Tracker::new(req_ctx));
 
         self.read_pool
-            .future_execute(priority, move |ctxd| {
-                tracker.attach_ctxd(ctxd);
-
+            .spawn(priority, move || {
+                tracker.init_current_stage();
                 Self::handle_stream_request_impl(engine, tracker, handler_builder) // Stream<Resp, Error>
                     .then(Ok::<_, mpsc::SendError<_>>) // Stream<Result<Resp, Error>, MpscError>
                     .forward(tx)
             })
             .map_err(|_| Error::Full)
-            .and_then(move |cpu_future| {
+            .and_then(move |_| {
                 // Keep running stream producer
-                cpu_future.forget();
+                // cpu_future.forget();
```

Comment:
PTAL @hicqu

Reply:
You can remove the line directly. It's OK because `spawn` has polled it internally. BTW I prefer to write
```rust
self.read_pool.spawn(...)?;
Ok(rx.then(|r| r.unwrap()))
```
to make it more clear that `spawn` returns a `Result<()>`.